cmake/SeastarDependencies.cmake

```diff
@@ -141,6 +141,9 @@ macro (seastar_find_dependencies)
   seastar_set_dep_args (rt REQUIRED)
   seastar_set_dep_args (numactl
     OPTION ${Seastar_NUMA})
+  seastar_set_dep_args (Sanitizers
+    COMPONENTS
+      ${Seastar_SANITIZERS})
```

Comment:
please note, this file is installed and redistributed by seastar. it is evaluated when a Seastar application consumes the Seastar library via CMake's `find_package(Seastar)`.

there is a potential issue with the current configuration:

1. At the time this file is processed, `Seastar_SANITIZERS` is empty.
1. This may result in an incomplete Seastar CMake configuration for the parent project.
1. If the user doesn't have sanitizer libraries installed, `find_package(Seastar)` still succeeds.
1. However, Seastar libraries compiled with sanitizers enabled may fail to configure or link due to the missing sanitizer libraries, as what we actually have in this file is

   ```cmake
   find_package(Sanitizers COMPONENTS)
   ```

Reply:
i am not sure what is the best way to address this. but there are at least two approaches:

- create a separate `cmake/SeastarDependencies.cmake.in`, and generate the cmake file using `configure_file()`, so we can have a different `SeastarDependencies.cmake` for each mode, and then `install()` it
- check all the valid combinations:

  - `address` + `undefined_behavior` 
  - `thread`